node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala

```diff
@@ -124,7 +125,14 @@ object WebApiRoutes {
         req.handle[DeployRequest, String](webApi.deploy)
 
       case req @ POST -> Root / "explore-deploy" =>
-        req.handle[String, ExploratoryDeployResponse](webApi.exploratoryDeploy)
+        req.handle[String, ExploratoryDeployResponse](
+          term => webApi.exploratoryDeploy(term, "", true)
+        )
+
+      case req @ POST -> Root / "v2" / "explore-deploy" =>
+        req.handle[ExploreDeployRequest, ExploratoryDeployResponse](
+          req => webApi.exploratoryDeploy(req.term, req.blockHash, req.isPostStateHash)
```

Comment:
We want to have this method in v1 also. I see this as an extension. Maybe we can call it `explore-deploy-by-block-hash` for clear separation from last finalized variant.